### PR TITLE
ISSUE-871 Backend should clean expired FCM subscriptions

### PR DESCRIPTION
**Pull request by quantranhong1999**

### CODE REVIEW

File: tmail-backend/jmap/extensions/src/main/scala/com/linagora/tmail/james/jmap/firebase/CleanExpiredFcmSubscriptionService.scala [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+import reactor.core.scala.publisher.{SFlux, SMono}
+
+object CleanExpiredFcmSubscriptionService {
+  private val LOGGER: Logger = LoggerFactory.getLogger(classOf[CleanExpiredFcmSubscriptionService])
+}
+
+class CleanExpiredFcmSubscriptionService @Inject()(subscriptionRepository: FirebaseSubscriptionRepository,
+                                                   usersRepository: UsersRepository,
+                                                   clock: Clock) extends Startable {
+  def cleanUp(): SMono[Int] = {
```
Comment (chibenwa) [inline tmail-backend/jmap/extensions/src/main/scala/com/linagora/tmail/james/jmap/firebase/CleanExpiredFcmSubscriptionService.scala:22]: If I do not reboot james, the cleanup is not done...

File: tmail-backend/jmap/extensions/src/main/scala/com/linagora/tmail/james/jmap/method/FirebaseSubscriptionGetMethod.scala [hunk 2]
```diff
@@ -45,13 +49,21 @@ class FirebaseSubscriptionGetMethod @Inject()()(val serializer: FirebaseSubscrip
...
             .map(_.get)
             .toSet.asJava))
-    }
+    })
+      .flatMap(subscription => {
+        if (isInThePast(subscription.expires, clock)) {
+          SMono(repository.revoke(username, subscription.id))
+            .`then`(SMono.empty)
+        } else {
+          SMono.fromCallable(() => subscription)
```
Comment (chibenwa) [inline tmail-backend/jmap/extensions/src/main/scala/com/linagora/tmail/james/jmap/method/FirebaseSubscriptionGetMethod.scala:66]:
```suggestion
          SMono.just(subscription)
```

Comment (quantranhong1999) [reply tmail-backend/jmap/extensions/src/main/scala/com/linagora/tmail/james/jmap/method/FirebaseSubscriptionGetMethod.scala:66]: `SMono.just` is an eager evaluation. I want a lazy evaluation here.

Comment (chibenwa) [reply tmail-backend/jmap/extensions/src/main/scala/com/linagora/tmail/james/jmap/method/FirebaseSubscriptionGetMethod.scala:66]: Isnt subscription already evaluated?

Comment (quantranhong1999) [reply tmail-backend/jmap/extensions/src/main/scala/com/linagora/tmail/james/jmap/method/FirebaseSubscriptionGetMethod.scala:66]: I mean the `SMono` variable would be eagerly created even if the flow does not fall into the else block, which is useless, no?

Comment (chibenwa) [reply tmail-backend/jmap/extensions/src/main/scala/com/linagora/tmail/james/jmap/method/FirebaseSubscriptionGetMethod.scala:66]: I think there is not difference for serving a pojo between the two.

**Review by Arsnael — APPROVED** [review]

**Pull request merged**